scripts/tests/chiptest/yamltest_with_chip_repl_tester.py

```diff
@@ -88,28 +88,31 @@ def main(setup_code, yaml_path, node_id):
         dev_ctrl = ca_list[0].adminList[0].NewController()
         dev_ctrl.CommissionWithCode(setup_code, node_id)
 
-        # Creating Cluster definition.
-        cluster_xml_filenames = glob.glob(_CLUSTER_XML_DIRECTORY_PATH + '/*/*.xml', recursive=False)
-        cluster_xml_filenames.sort(key=functools.cmp_to_key(_sort_with_global_attribute_first))
-        sources = [ParseSource(source=name) for name in cluster_xml_filenames]
-        clusters_definitions = SpecDefinitions(sources)
-
-        # Parsing YAML test and setting up chip-repl yamltests runner.
-        yaml = TestParser(yaml_path, None, clusters_definitions)
-        runner = ReplTestRunner(clusters_definitions, certificate_authority_manager, dev_ctrl)
-
-        # Executing and validating test
-        for test_step in yaml.tests:
-            test_action = runner.encode(test_step)
-            # TODO if test_action is None we should see if it is a pseudo cluster.
-            if test_action is not None:
-                response = runner.execute(test_action)
-                decoded_response = runner.decode(response)
-                post_processing_result = test_step.post_process_response(decoded_response)
-                if not post_processing_result.is_success():
-                    exit(-2)
-            else:
-                exit(-2)
+        try:
+            # Creating Cluster definition.
+            cluster_xml_filenames = glob.glob(_CLUSTER_XML_DIRECTORY_PATH + '/*/*.xml', recursive=False)
+            cluster_xml_filenames.sort(key=functools.cmp_to_key(_sort_with_global_attribute_first))
+            sources = [ParseSource(source=name) for name in cluster_xml_filenames]
+            clusters_definitions = SpecDefinitions(sources)
+
+            # Parsing YAML test and setting up chip-repl yamltests runner.
+            yaml = TestParser(yaml_path, None, clusters_definitions)
+            runner = ReplTestRunner(clusters_definitions, certificate_authority_manager, dev_ctrl)
+
+            # Executing and validating test
+            for test_step in yaml.tests:
+                test_action = runner.encode(test_step)
+                # TODO if test_action is None we should see if it is a pseudo cluster.
+                if test_action is not None:
+                    response = runner.execute(test_action)
+                    decoded_response = runner.decode(response)
+                    post_processing_result = test_step.post_process_response(decoded_response)
+                    if not post_processing_result.is_success():
+                        raise Exception(f'Test step failed {test_step.label}')
+                else:
+                    raise Exception(f'Failed to encode test step {test_step.label}')
+        except Exception:
+            exit(-2)
```

Comment:
should we print a traceback here to debug? I am afraid exit will lose the exception message and stack (or is that not true?)